src/components/MapView/index.tsx

```diff
@@ -242,59 +242,68 @@ function MapView({ classes }: MapViewProps) {
     }
 
     if (
-      (!hazardLayerId && !baselineLayerId) ||
+      (!hazardLayerIds && !baselineLayerId) ||
       Object.keys(serverAvailableDates).length === 0
     ) {
       return;
     }
 
     const selectedLayersIds: LayerKey[] = selectedLayers.map(layer => layer.id);
 
-    // Check for layers in url and add them.
-    [hazardLayerId, baselineLayerId].forEach(id => {
-      if (!id || selectedLayersIds.includes(id as LayerKey)) {
-        return;
-      }
+    const hazardLayersArray =
+      hazardLayerIds !== null ? hazardLayerIds.split(',') : [];
 
-      if (Object.keys(LayerDefinitions).includes(id)) {
-        const layer = LayerDefinitions[id as LayerKey];
-        dispatch(addLayer(layer));
+    const urlLayerIds = [
+      ...hazardLayersArray,
+      ...(baselineLayerId === null ? [] : baselineLayerId),
+    ];
 
-        if (selectedDate && !urlDate) {
-          updateHistory(
-            'date',
-            moment(selectedDate).format(DEFAULT_DATE_FORMAT),
-          );
-        }
-      } else {
-        dispatch(
-          addNotification({
-            message: `Invalid layer identifier: ${id}`,
-            type: 'error',
-          }),
-        );
-      }
+    // Check for invalid layer ids.
+    const layerDefinitionIds = Object.keys(LayerDefinitions);
+    const invalidLayersIds = urlLayerIds.filter(
+      layerId => layerDefinitionIds.includes(layerId) === false,
+    );
+
+    if (invalidLayersIds.length > 0) {
+      const invalidLayersStr = invalidLayersIds.join(',');
+
+      dispatch(
+        addNotification({
+          message: `Invalid layer identifier(s): ${invalidLayersStr}`,
+          type: 'error',
+        }),
+      );
+
+      return;
+    }
+
+    // Check for layers that have not been included.
+    const missingLayers = urlLayerIds.filter(
+      layerId => selectedLayersIds.includes(layerId as LayerKey) === false,
+    );
+
+    missingLayers.forEach(layerId => {
+      const layer = LayerDefinitions[layerId as LayerKey];
+      dispatch(addLayer(layer));
     });
 
-    if (
-      urlDate &&
-      moment(urlDate).valueOf() !== selectedDate &&
-      selectedLayersIds.includes(hazardLayerId as LayerKey)
-    ) {
-      const dateInt = moment(urlDate).valueOf();
-      if (Number.isNaN(dateInt)) {
-        dispatch(
-          addNotification({
-            message: 'Invalid date found. Using most recent date',
-            type: 'warning',
-          }),
-        );
-      } else {
-        dispatch(updateDateRange({ startDate: dateInt }));
-      }
+    const dateInt = moment(urlDate).valueOf();
+
+    if (urlDate === null || dateInt === selectedDate) {
```

Comment:
`if (!urlDate || dateInt === selectedDate)`

Reply:
fixed in commit 244f3695edd712317dd482073e3fdad250bb0d2f